actors/builtin/multisig/multisig_test.go

```diff
@@ -75,6 +76,111 @@ func TestConstruction(t *testing.T) {
 	})
 }
 
+func TestVesting(t *testing.T) {
+	actor := msActorHarness{multisig.MultiSigActor{}, t}
+
+	receiver := newIDAddr(t, 100)
+	anne := newIDAddr(t, 101)
+	bob := newIDAddr(t, 102)
+	charlie := newIDAddr(t, 103)
+	darlene := newIDAddr(t, 103)
+
+	const unlockDuration = 10
+	var multisigInitialBalance = abi.NewTokenAmount(100)
+	var nilParams = abi.MethodParams{}
+
+	builder := mock.NewBuilder(context.Background(), receiver).
+		WithCaller(builtin.InitActorAddr, builtin.InitActorCodeID).
+		WithEpoch(0).
+		// balance 0: current balance of the actor. receive: 100 the amount the multisig actor will be initalized with -- InitialBalance
+		WithBalance(abi.NewTokenAmount(0), multisigInitialBalance)
+
+	t.Run("happy path full vesting", func(t *testing.T) {
+		rt := builder.Build(t)
+
+		actor.constructAndVerify(rt, 2, unlockDuration, []addr.Address{anne, bob, charlie}...)
+
+		// anne proposes that darlene receives `multisgiInitialBalance` FIL.
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.propose(rt, darlene, multisigInitialBalance, builtin.MethodSend, nilParams)
+		rt.Verify()
+
+		// set the current balance of the multisig actor to its InitialBalance amount
+		rt.SetBalance(multisigInitialBalance)
+		// Advance the epoch s.t. all funds are unlocked.
+		rt.SetEpoch(0 + unlockDuration)
+		// bob approves annes transaction
+		rt.SetCaller(bob, builtin.AccountActorCodeID)
+		// expect darlene to receive the transaction proposed by anne.
+		rt.ExpectSend(darlene, builtin.MethodSend, nilParams, multisigInitialBalance, nil, exitcode.Ok)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.approve(rt, 0)
+		rt.Verify()
+
+	})
+
+	t.Run("partial vesting", func(t *testing.T) {
+		rt := builder.Build(t)
+
+		actor.constructAndVerify(rt, 2, 10, []addr.Address{anne, bob, charlie}...)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.propose(rt, darlene, big.Div(multisigInitialBalance, big.NewInt(2)), builtin.MethodSend, nilParams)
+		rt.Verify()
+
+		// set the current balance of the multisig actor to its InitialBalance amount
+		rt.SetBalance(multisigInitialBalance)
+		rt.SetEpoch(0 + unlockDuration/2)
+		rt.SetCaller(bob, builtin.AccountActorCodeID)
+		rt.ExpectSend(darlene, builtin.MethodSend, nilParams, big.Div(multisigInitialBalance, big.NewInt(2)), nil, exitcode.Ok)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.approve(rt, 0)
+		rt.Verify()
+
+	})
+
+	t.Run("fail to propose a transaction greater than locked amount", func(t *testing.T) {
```

Comment:
the wording here is a bit off. You should be able to _propose_ a transaction spending the locked amount, but not submit the final approval for it. We should exercise that: roll it into the partial vesting test above: first try and fail to spend the whole balance.

this one might say "propose and autoapprove transaction above locked amount" (and you should also test that below the locked amount is ok.